lib/puppet/pal/pal_impl.rb

```diff
@@ -49,6 +49,7 @@ module Pal
   #   If given at the environment level, the facts given here are merged with higher priority.
   # @param variables [Hash] optional map of fully qualified variable name to value. If given at the environment level, the variables
   #   given here are merged with higher priority.
+  # @param disable_facts [Boolean] when true, the $facts, $server_facts, and $trusted variables for the scope are not set.
```

Comment:
This name is a little confusing. For one, it would be clearer if it _enabled_ a feature instead of _disabling_ it. But it also reads like it disables the entire _concept_ of facts. Maybe something like `set_local_facts: true`?